vcsclient/fs.go

```diff
@@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
 	return &fwr, nil
 }
 
+// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
+// shouldRecurse is set to true, it will recurse into all sub-folders and return the
+// full sub-tree.
+func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
```

Comment:
I'm not going to suggest you use it here, but I do have a package for walking the correct `vfs.FileSystem` type that you have.

See https://godoc.org/github.com/shurcooL/go/vfs/godocfs/vfsutil#example-Walk.